### PR TITLE
Spherefix & Reverse Effect & Delay Time to start

### PR DESCRIPTION
**Pull request by telmiuse**

### CODE REVIEW

**Comment by telmiuse** [comment]
Gw9Wz2sMgm

**Review by JaimeBea — APPROVED** [review]

**Review by FrancPS — APPROVED** [review]

**Comment by telmiuse** [comment]
The delay time in Loop Just work in the first time, once stard does not affect it .

**Review by JaimeBea — APPROVED** [review]

File: Project/Source/Components/ComponentParticleSystem.cpp [hunk 1]
```diff
@@ -33,6 +33,9 @@
 #define JSON_TAG_BILLBOARDTYPE "BillboardType"
 #define JSON_TAG_TEXTURE_TEXTUREID "TextureId"
 
+#define JSON_TAG_TEXTURE_DISTANCEREVERSE "distanceReverse"
+#define JSON_TAG_TEXTURE_REVERSEEFFECT "reverseEffect"
```
Comment (Jordi-Gil) [inline Project/Source/Components/ComponentParticleSystem.cpp:37]: Tag in lowcase

File: Project/Source/Components/ComponentParticleSystem.cpp [hunk 2]
```diff
@@ -256,16 +287,21 @@ void ComponentParticleSystem::Load(JsonValue jComponent) {
...
-	randomDirection = jComponent[JSON_TAG_ISRANDOMDIRECTION];
 	scale = jComponent[JSON_TAG_SCALEPARTICLE];
 	maxParticles = jComponent[JSON_TAG_MAXPARTICLE];
 	velocity = jComponent[JSON_TAG_VELOCITY];
 	particleLife = jComponent[JSON_TAG_LIFE];
 	sizeOverTime = jComponent[JSON_TAG_SIZEOVERTIME];
 	scaleFactor = jComponent[JSON_TAG_SCALEFACTOR];
+	startDelay = jComponent[JSON_TAG_STARTDELAYTIME];
+	reverseEffect = jComponent[JSON_TAG_TEXTURE_DISTANCEREVERSE];
+	distanceReverse = jComponent[JSON_TAG_TEXTURE_REVERSEEFFECT];
```
Comment (Jordi-Gil) [inline Project/Source/Components/ComponentParticleSystem.cpp:304]: reverseEffect and distanceReverse swap

File: Project/Source/Components/ComponentParticleSystem.cpp [hunk 3]
```diff
@@ -292,17 +328,22 @@ void ComponentParticleSystem::Save(JsonValue jComponent) const {
...
-	jComponent[JSON_TAG_ISRANDOMDIRECTION] = randomDirection;
 	jComponent[JSON_TAG_SCALEPARTICLE] = scale;
 	jComponent[JSON_TAG_MAXPARTICLE] = maxParticles;
 	jComponent[JSON_TAG_VELOCITY] = velocity;
 	jComponent[JSON_TAG_LIFE] = particleLife;
 	jComponent[JSON_TAG_SIZEOVERTIME] = sizeOverTime;
 	jComponent[JSON_TAG_SCALEFACTOR] = scaleFactor;
 
+	jComponent[JSON_TAG_TEXTURE_DISTANCEREVERSE] = reverseEffect;
+	jComponent[JSON_TAG_TEXTURE_REVERSEEFFECT] = distanceReverse;
```
Comment (Jordi-Gil) [inline Project/Source/Components/ComponentParticleSystem.cpp:345]: reverseEffect and distanceReverse swap

File: Project/Source/Components/ComponentParticleSystem.h [hunk 4]
```diff
@@ -81,20 +87,25 @@ class ComponentParticleSystem : public Component {
...
+	float restDelayTime = 0;
+
 	unsigned maxParticles = 100;
 	unsigned particleSpawned = 0;
 	float velocity = 0.1f;
 	float particleLife = 5;
 	float scaleFactor = 0;
-
-
+	float coneRadiusUp = 1;
```
Comment (Jordi-Gil) [inline Project/Source/Components/ComponentParticleSystem.h:107]: float initialize as an integer. Set 1.0f preferred

**Review by Jordi-Gil — CHANGES_REQUESTED** [review]

**Review by JaimeBea — APPROVED** [review]

**Review by Jordi-Gil — APPROVED** [review]

**Review by lowysole — DISMISSED** [review]

**Pull request merged**